model/src/main/kotlin/OrtResult.kt

```diff
@@ -247,6 +247,12 @@ data class OrtResult(
             ?: getPackage(id)?.pkg?.declaredLicensesProcessed?.allLicenses?.toSortedSet()
             ?: sortedSetOf()
 
+    /**
+     * Return the detected licenses for the given package [id]. If [id] is not found an empty set is returned.
+     */
+    fun getDetectedLicensesForId(id: Identifier): SortedSet<SpdxExpression> =
+        getScanResultsForId(id).flatMapTo(sortedSetOf((compareBy { it.toString() }))) { it.summary.licenses }
```

Comment:
FYI, I just filed https://github.com/oss-review-toolkit/ort/pull/3569. Maybe we can merge that first, and then rebase this to avoid the need for `compareBy { it.toString() }` here?